js/ui/viewSelector.js

```diff
@@ -113,6 +119,84 @@ const ShowOverviewAction = new Lang.Class({
     }
 });
 
+const ViewsDisplayLayout = new Lang.Class({
+    Name: 'ViewsDisplayLayout',
+    Extends: Clutter.BinLayout,
+
+    _init: function(appDisplayActor) {
+        this.parent();
+
+        this._appDisplayActor = appDisplayActor;
+        this._appDisplayActor.connect('style-changed', Lang.bind(this, this._onStyleChanged));
+    },
+
+    _onStyleChanged: function() {
+        this.layout_changed();
+    }
+});
+
+const ViewsDisplayContainer = new Lang.Class({
+    Name: 'ViewsDisplayContainer',
+    Extends: St.Widget,
+    Signals: {
+        'views-page-changed': { }
+    },
+
+    _init: function(appDisplay) {
+        this._appDisplay = appDisplay;
+        this._activePage = ViewsDisplayPage.APP_GRID;
+
+        this.parent({ layout_manager: new ViewsDisplayLayout(this._appDisplay.actor),
+                      x_expand: true,
+                      y_expand: true });
+
+        this.add_actor(this._appDisplay.actor);
+    },
+
+    showPage: function(page) {
```

Comment:
Actually I don't think we need to emit any signal, at least not for now. It was used in the old shell to call `updateBackgroundShade()` from the overview, but we're not doing that for now here, so the emission can be removed altogether.